scripts/release-commit.rb

```diff
@@ -12,6 +12,25 @@
 # Functions
 #
 
+def bump_cargo_version(version)
+  # Cargo.toml
+  content = File.read("#{ROOT_DIR}/Cargo.toml")
+  new_content = bump_version(content, version)
+  File.write("#{ROOT_DIR}/Cargo.toml", new_content)
+
+  # Cargo.lock
+  content = File.read("#{ROOT_DIR}/Cargo.lock")
```

Comment:
I believe this file should only be modified by cargo.lock for forwards compat reasons?

Reply:
Previously we needed to run `cargo check` to update this. This felt like a lighter weight option.

Reply:
Yeah, running `cargo check` is the proper way. I know in the past we have gotten new versions of rust that changed how it generated the lock file and this is something that is not guaranteed to not change. So if we do anything we should just run `cargo check` or not bump in CI and do it via a PR that updates cargo.toml and has the user run cargo check locally.

Reply:
This isn't done during CI.

Reply:
Ah ok yeah, then it should bump `Cargo.toml` and then run `cargo check` to ensure that code actually compiles. I think to me that sounds right. Happy to hear other solutions 😄

Reply:
I guess I don't understand why `cargo check` is necessary for this small change? The Cargo files should already be valid, bumping the internal Vector version should not change anything about Vectors environment that would then make the Cargo files invalid. Also, I can't event run `cargo check` locally:

```
--- stderr
Building and linking librdkafka statically
Running command: "./configure --disable-sasl --enable-ssl --disable-lz4 --enable-static" in dir: librdkafka
thread 'main' panicked at 'Command failed with error: No such file or directory (os error 2)', /Users/benjohnson/.cargo/registry/src/github.com-1ecc6299db9ec823/rdkafka-sys-0.11.6-1/build.rs:25:21
```

Reply:
You need to install the required libraries to build vector. By updating Cargo.toml, it updates cargo.lock for you when you run `cargo check` with the proper changes. You can think of the loc as the generated doc files and `cargo check` running `make generate`. For now though you can run `cargo check --no-default-features` and it won't build any of our C++/C deps.

Reply:
I'm not super comfortable overwriting `Cargo.lock` ourselves, especially with a regex and no validation. I also get that `cargo check` is more than we need.

Instead, let's do the regex for `Cargo.toml` and then use the following, which will update `Cargo.lock` and validate that we set the version we intended to (i.e. the regex didn't miss):

```
cargo metadata --format-version 1 | jq -r '.packages[] | select(.name == "vector") | .version'
```

That shouldn't require actually building any code, but it will make sure we didn't accidentally mess anything up.